wavecrestweb/kenworthy-machine
# #KM-13/request a quote section

## Description
This PR adds the request a quote section to the home page. The text formatting is still a WIP as I'm still trying to figure out how to best parse rich-text formatted text from the CMS. I'd love some help with this! The form is intentionally not functional. I believe we should address that in a separate PR as it requires research into email.


## Related Tickets & Documents
Closes #KM-13


## Acceptance Criteria

- [x] The form includes fields for name, email, and industry type.
- [x] Each field has placeholder text to indicate the information required.
- [x] The form design and input fields are visually aligned with the overall website aesthetic.
- [ ] The form provides an intuitive experience on both desktop and mobile devices.
- [ ] Inputted data is validated for format and completeness before submission.




## What type of PR is this? (check all applicable)


|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |
|   ✓ | 🎨 Style                   |
|     | :hammer: Refactoring       |
|     | 🔥 Performance Improvements|
|     | :bug: Bug fix              |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |


## Mobile & Desktop Screenshots/Recordings
![Screenshot 2024-02-06 at 11 57 34 AM](https://github.com/wavecrestweb/kenworthy-machine/assets/115492619/e130b2cc-2dc3-4ae6-bf14-21c7954e88d8)



## What I learned
I'm still trying to figure out how to best parse markdown from the CMS in the components. I think how it's rendered in the component impacts how effective the visual editor is. Right now, text changes to this component in the visual editor render but formatting changes like line breaks and bold do not.

## Testing steps

1. run `npm i` if you haven't installed the CMS related packages yet
2. `git switch KM-13/feature-request-quote`
3. navigate to localhost:3000/admin
4. inspect the request a quote section on the home page
5. inspect the component in code, test to see how to better render the 3 text blocks under 'send us your request' so the line break shows up where it's needed.

## Added to documentation?

|     | Type                       |
| --- | -------------------------- |
|     | 📜 README.md    |
| ✓ | 🙅 no documentation needed    |

## What gif best describes this PR?
How I feel about the home page being almost done!
  ![getting closer](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExeWlzOXM4ZzVneDJiYzVmMm5mbGFvMDBiY25kcmE1a2d2cDlvMzFnMSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/Z8lJWgEqMlvoVS9lzr/giphy.gif)
